tls/s2n_connection.c

```diff
@@ -526,6 +526,10 @@ int s2n_connection_set_config(struct s2n_connection *conn, struct s2n_config *co
     }
     conn->tickets_to_send = config->initial_tickets_to_send;
 
+    if (conn->psk_params.psk_list.len == 0) {
```

Comment:
It looks like the rest of the values are taken from the config, except in the `client_cert_auth_type_overridden` case. Should we add an `override` for PSKs as well? If a connection is reused, will it always keep the original config->psk_mode?